### ISSUE TITLE
Pager throws exception if git is not installed

### ISSUE THREAD

**Issue by slowbro** [piotrmurach/tty-pager#9]
I have a feeling this is due to not checking if there is a 'git' executable before doing over this section:

https://github.com/piotrmurach/tty-pager/blob/8e2e266b1cc0dc8fa434236087131e7203c26ff7/lib/tty/pager/system.rb#L100-L104
The solution was to just install git, but it would be great if the lib checked for git first.

**Comment by piotrmurach** [comment]
Hi Katelyn,

Thank you for trying this library out!

I agree with you that checking for git executable is important but as far as this library goes that's exactly what happens. The `executables` is an array of binaries but the real usage happens here https://github.com/piotrmurach/tty-pager/blob/master/lib/tty/pager/system.rb#L47 where each command is checked for existence. This check is done based on `PATH` environment variable which in your case must have contained `git` even when it was not installed. Is this possible?

**Comment by slowbro** [comment]
Ah, that's odd! I already installed it, but let me see if I can replicate the issue. It shouldn't have been in my PATH. This is on FreeBSD, too, so maybe that command behaves differently. I'll update later today with what I can find!

Also, great libs. We just started using tty-* for all our internal CLI utils, and it's made them 1000% prettier.

**Comment by slowbro** [comment]
Alright, here is an example of it not working. 

```ruby
#!/usr/bin/env ruby
require 'tty-pager'

pager = TTY::Pager.new

info = "info"*600000

pager.page(info)
```

```shell
$ ./test.rb
/usr/local/lib/ruby/gems/2.3/gems/tty-pager-0.10.0/lib/tty/pager/system.rb:102:in ``': No such file or directory - git (Errno::ENOENT)
        from /usr/local/lib/ruby/gems/2.3/gems/tty-pager-0.10.0/lib/tty/pager/system.rb:102:in `executables'
        from /usr/local/lib/ruby/gems/2.3/gems/tty-pager-0.10.0/lib/tty/pager/system.rb:44:in `available'
        from /usr/local/lib/ruby/gems/2.3/gems/tty-pager-0.10.0/lib/tty/pager/system.rb:62:in `available?'
        from /usr/local/lib/ruby/gems/2.3/gems/tty-pager-0.10.0/lib/tty/pager.rb:93:in `find_available'
        from /usr/local/lib/ruby/gems/2.3/gems/tty-pager-0.10.0/lib/tty/pager.rb:31:in `initialize'
        from ./test.rb:4:in `new'
        from ./test.rb:4:in `<main>'
```

```
[katelyn@holp ~]$ sudo pkg info | grep git
[katelyn@holp ~]$ which git
[katelyn@holp ~]$ echo $PATH
/sbin:/bin:/usr/sbin:/usr/bin:/usr/games:/usr/local/sbin:/usr/local/bin:/home/k/katelyn/bin
```

I verified that git is not in any of the directories in my PATH.

**Comment by piotrmurach** [comment]
That would be great help! 

Thank you for using tty-* gems, that makes it more worthwhile developing them and I'm also glad they make your CLI utils look great ✨

**Comment by piotrmurach** [comment]
Thank you for looking into this. I agree that the cause is probably the backticks execution.

Would you have time to submit PR fixing this? A test case trapping this bug would be great as well.

**Comment by slowbro** [comment]
PR#1 (woot!) is in now.

**Comment by slowbro** [comment]
Whoops, i messed up that PR.. did it on my own project :| actually submitted, now

**Comment by piotrmurach** [comment]
Released `v0.11.0` with your fix!
